docs/isl-2-0/spec.md

```diff
@@ -0,0 +1,762 @@
+Whether a particular value matches a specified value is governed by the equivalence rules defined by the Ion data model.
+For example, `1.230` is not valid for `valid_values: [1.23]`, as it has a different precision.
+While these may be mathematically equal values, they are not _equivalent data_.
+In particular, note that `nan` is valid for `valid_values: [nan]` because in the Ion data model, all valid Ion encodings of `nan` are _equivalent data_.
+
+{% include note.html type="note" content="
+`valid_values` only matches a single Ion value.
+Recall that a `document` is not a single Ion value, but rather a stream of values, so a `document` is always invalid for the `valid_values` constraint.
+" %}
+
+For numeric and timestamp types, `valid_values` may optionally be defined as a range.
+Numeric and timestamp ranges follow all the rules specified in [§Ranges](#ranges).
+
+A `number` range includes all values of any numeric type (i.e. `float`, `int`, and `decimal`) that fall within that range mathematically.
+When testing a value for inclusion, the range bounds and the value that is being tested will all be converted to `decimal` values for the sake of comparison.
+`number` ranges do not include values of any other type.
+The `float` type includes [special non-number values](https://amzn.github.io/ion-docs/docs/float.html#special-values) (`nan`, `+inf`, and `-inf`).
+Attempting to use a non-number value as the bound of a number range will result in an error.
+When tested for inclusion in a number range, the non-number float values are always outside the bounds of the number range, even when one of the range bounds is `min` or `max`.
+Number-typed `null` values are never valid for any number range.
+
+A `timestamp` range includes any `timestamp` that falls between the minimum and maximum in chronological order.
+All [`timestamp` values](https://amzn.github.io/ion-docs/docs/spec.html#timestamp) represent an instant in time.
+A `timestamp` with limited precision (for example, a year-only timestamp like `2007T`) maps to an instant by assuming that all unspecified time unit fields are effectively zero (or one for the month and day units). 
+`2007T` would map to the instant represented by `2007-01-01T00:00.000-00:00`.
+Note that `timestamp` values that do not have a time component (that is `YYYYT`, `YYYY-MMT`, and `YYYY-MM-DDT` timestamps) have an unknown offset (`-00:00`).
+Timestamps that have an unknown offset are UTC timestamps that make no assertion about the offset in which they occurred.
+The value `null.timestamp` is never in the bounds of a timestamp range, and `timestamp` ranges do not include values of any other type.
+
+# Open Content
+
+{% include grammar-element.md productions="user_content_declaration,user_content_declaration_field" %}
+
+The Ion Schema 2.0 specification allows users to insert additional content into a schema document that is not part of the Ion Schema Language.
```

Comment:
I recommend restructuring this sentence to improve clarity. Since it's the "additional content", not the "schema document" that is "not part of the Ion schema language", I'd suggest:

> The Ion Schema 2.0 specification allows users to insert additional content that is not part of the Ion Schema language into a schema document.